### PR TITLE
manifold: init at 2.5.1-unstable-2024-08-18

### PR DESCRIPTION
**Pull request by hzeller**
## Description of changes

Adding manifold library, a very fast geometry library.

Last release was 2.5.1, but that still requires a heavy NVIDIA thrust dependency, the latest head does not need that anymore, thus starting with head as 2.5.1-unstable.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by hzeller** [comment]
@pca006132 : adding you as co-maintainer of this package as you're actively involved in manifold development.

File: pkgs/by-name/ma/manifold/package.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    gtest
+    glm
+    tbb
+    clipper2
+  ];
+
+  cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=Release"
+    "-DBUILD_SHARED_LIBS=ON"
+    "-DMANIFOLD_TEST=ON"
```
Comment (pca006132) [inline pkgs/by-name/ma/manifold/package.nix:35]: I think `MANIFOLD_TEST` is default to `ON` already?

**Comment by pca006132** [comment]
Thanks! I was thinking about adding a package to nix when we finish v3.0, which contains the thrust removal as well as other things. Forgot that we can also add unstable :)

**Comment by pca006132** [comment]
Btw, maybe we can package the python package as well? Not very sure about the procedure of packaging that.

File: pkgs/by-name/ma/manifold/package.nix (hunk 1 above)
Comment (hzeller) [reply pkgs/by-name/ma/manifold/package.nix:35]: I wanted it to be explicit, as `doCheck = true;` is set. This makes it independent of the defaults in the CMakeLists.

**Comment by hzeller** [comment]
> Btw, maybe we can package the python package as well? Not very sure about the procedure of packaging that.

I don't use Python so I don't really know how that is done best and what Python users expect. So i'd see this as PR a starting point for someone adding the necessary Python things later.

**Comment by hzeller** [comment]
Just found out that also `pkg-config` is needed while building for cmake to find tbb. Updated PR.

**Comment by hzeller** [comment]
> I think the `pkg-config` is needed due to `tbb` being a bit old. If we use [tbb_2021_11](https://search.nixos.org/packages?channel=24.05&show=tbb_2021_11&from=0&size=50&sort=relevance&type=packages&query=tbb) we can probably skip `pkg-config`.

Thanks, indeed, the newer tbb actually works natively with cmake. Updated PR.

**Review by pca006132 — APPROVED** [review]

**Comment by hzeller** [comment]
FYI, updated to latest head to include the manifold *.pc file. You probably have to re-approve due to the change.

**Review by pca006132 — APPROVED** [review]

**Review by pca006132 — APPROVED** [review]

**Comment by hzeller** [comment]
Now we just need someone with merge permissions...

**Comment by pca006132** [comment]
@7c6f434c can you help us merge this?

**Comment by 7c6f434c** [comment]
Not sure what to do with apparently macOS-versions-specific issue (given that I don't use macOS), I guess it's fine to leave as is so that people on fresh macOS can try…

The package code looks reasonable to me (I have no idea what the package does and how it should work, of course)

**Pull request merged**